Kernel/WorkQueue.cpp

```diff
@@ -12,18 +12,11 @@
 
 namespace Kernel {
 
-WorkQueue* g_io_work;
-
-UNMAP_AFTER_INIT void WorkQueue::initialize()
-{
-    g_io_work = new WorkQueue("IO WorkQueue");
-}
-
-UNMAP_AFTER_INIT WorkQueue::WorkQueue(const char* name)
+WorkQueue::WorkQueue(String&& name)
 {
     RefPtr<Thread> thread;
-    Process::create_kernel_process(thread, name, [this] {
-        for (;;) {
+    Process::create_kernel_process(thread, move(name), [this] {
```

Comment:
Isn't creating a kernel process per work queue a bit heavy handed? 
For example one for each AHCIPort could end up being alot...

Reply:
Hm good point, we should have one process but multiple threads only, good catch. This is a left-over from before, where we only had one WorkQueue for everything.